client/network/src/protocol/notifications/behaviour.rs

```diff
@@ -2034,6 +2154,44 @@ impl NetworkBehaviour for Notifications {
 			}
 		}
 
+		// poll commands from protocols
+		loop {
+			match futures::Stream::poll_next(Pin::new(&mut self.command_streams), cx) {
+				Poll::Ready(Some((set_id, command))) => match command {
+					NotificationCommand::SetHandshake(handshake) => {
+						self.set_notif_protocol_handshake(set_id.into(), handshake);
+					},
+					NotificationCommand::OpenSubstream(_peer) |
+					NotificationCommand::CloseSubstream(_peer) => {
+						todo!("substream control not implemented");
+					},
+				},
+				Poll::Ready(None) => {
+					error!(target: "sub-libp2p", "Protocol command streams have been shut down");
+					break
+				},
+				Poll::Pending => break,
+			}
+		}
+
+		while let Poll::Ready(Some((result, index))) =
+			self.pending_inbound_validations.poll_next_unpin(cx)
+		{
+			match result {
+				Ok(ValidationResult::Accept) => {
+					self.peerset_report_accept(index);
+				},
+				Ok(ValidationResult::Reject) => {
+					// TODO(aaro): remove connection from peerset
```

Comment:
I would do it inside `protocol_report_reject` after we have resolved the incoming index. We likely need two similar methods `peerset_report_(pre)reject` and `protocol_report_reject` with the latter differing only in notifying peerset with:
```rust
self.peerset.dropped(set_id, *peer_id, DropReason::Unknown);
```

Reply:
yeah that is the plan, I'll leave this comment unresolved until it's fixed

Reply:
Fixed in https://github.com/paritytech/substrate/pull/14197/commits/295a6a7d970ce62996685275fb3027015df06a71